public/app/features/manage-dashboards/components/ImportDashboardForm.tsx

```diff
@@ -0,0 +1,185 @@
+  };
+
+  onSubmit = () => {
+    this.props.saveDashboard(this.state.folderId);
+  };
+
+  onCancel = () => {
+    this.props.resetDashboard();
+  };
+
+  onTitleChange = (event: FormEvent<HTMLInputElement>) => {
+    this.props.changeDashboardTitle(event.currentTarget.value);
+  };
+
+  onFolderChange = ($folder: { title: string; id: number }) => {
+    this.setState({ folderId: $folder.id });
+  };
+
+  onUidChange = (event: FormEvent<HTMLInputElement>) => {
+    this.props.changeDashboardUid(event.currentTarget.value);
+  };
+
+  onUidReset = () => {
+    this.setState({ uidReset: true });
+  };
+
+  render() {
+    const { dashboard, inputs, meta, source, uidExists, uidError, nameExists, nameError } = this.props;
+    const { uidReset } = this.state;
+
+    const buttonVariant: ButtonVariant = uidExists || nameExists ? 'destructive' : 'primary';
+    const buttonText = uidExists || nameExists ? 'Import (Overwrite)' : 'Import';
+
+    return (
+      <>
+        {source === DashboardSource.Gcom && (
+          <div style={{ marginBottom: '24px' }}>
+            <div>
+              <Forms.Legend>
+                Importing Dashboard from{' '}
+                <a
+                  href={`https://grafana.com/dashboards/${dashboard.gnetId}`}
+                  className="external-link"
+                  target="_blank"
+                >
+                  Grafana.com
+                </a>
+              </Forms.Legend>
+            </div>
+            <table className="filter-table form-inline">
+              <tbody>
+                <tr>
+                  <td>Published by</td>
+                  <td>{meta.orgName}</td>
+                </tr>
+                <tr>
+                  <td>Updated on</td>
+                  <td>{dateTime(meta.updatedAt).format('YYYY-MM-DD HH:mm:ss')}</td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+        )}
+        <Forms.Form onSubmit={this.onSubmit}>
+          {() => {
+            return (
+              <>
+                <Forms.Legend className="section-heading">Options</Forms.Legend>
+                <Forms.Field label="Name" invalid={nameExists} error={nameError}>
+                  <Forms.Input size="md" type="text" value={dashboard.title} onChange={this.onTitleChange} />
+                </Forms.Field>
+                <Forms.Field label="Folder">
+                  <FolderPicker onChange={this.onFolderChange} useInNextGenForms={true} initialFolderId={0} />
+                </Forms.Field>
+                <Forms.Field
+                  label="Unique identifier (uid)"
+                  description="The unique identifier (uid) of a dashboard can be used for uniquely identify a dashboard between multiple Grafana installs.
+                The uid allows having consistent URL’s for accessing dashboards so changing the title of a dashboard will not break any
+                bookmarked links to that dashboard."
+                  invalid={uidExists}
```

Comment:
See https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Forms/Form.mdx#errors for how to handle Form's errors

Reply:
How would you handle this particular case for errors?